### ISSUE TITLE
project billable button does not work

### ISSUE THREAD

**Issue by hostingnuggets** [DefactoSoftware/Hours#338]
The billable button under a projet does not do anything, is this behaviour wanted?

**Comment by jurre** [comment]
I'm not sure I understand what you mean, this button marks a project as billable. Is this a UI issue? If so, which browser etc are you using?

**Comment by hostingnuggets** [comment]
yeah UI issue, if I click on the button nothing happens, afaik I should be able to trigger it on or off. I am using iceweasel 31.7.0 as browser on Debian 8. Also I was wondering if you can define somewhere the hourly rate for a project?

**Comment by fatboypunk** [comment]
Also goes wrong for Firefox I see

**Comment by jurre** [comment]
> Also I was wondering if you can define somewhere the hourly rate for a project?

Nope, it's just time registration. You can easily export the data and could cook something up in google docs/excel though.

**Comment by dkhgh** [comment]
@hostingnuggets this is because you first need to specify a __client__ before the toggle is active:

1. No client selected (toggle is inactive):
![screen shot 2015-07-01 at 11 10 28](https://cloud.githubusercontent.com/assets/4135610/8451219/e9069038-1fe1-11e5-948d-ec1216dcb81a.png)

2. Client selected (toggle is active):
![screen shot 2015-07-01 at 11 10 41](https://cloud.githubusercontent.com/assets/4135610/8451227/ffd5a754-1fe1-11e5-95b7-5c2beed8a1ca.png)

3. Billable = true: 
![screen shot 2015-07-01 at 11 10 48](https://cloud.githubusercontent.com/assets/4135610/8451234/120ac38c-1fe2-11e5-86e1-7a0bd80d81d0.png)

So what I think we can take from this is that the UI here lacks clarity. 
* The difference between the toggle being `inactive` of `active` is negligible. 
* We don't explain __why__ the toggle is inactive.
* Why can't a project be billable without a client? (maybe you want to add the client later?)
* Why can't we add a new client inline here? Instead of first having to create a client at `/clients`.

Also some explanatory stuff and or inline validation would help here..

Any thoughts?

**Comment by bingxie** [comment]
@dkhgh your suggestions are very good. At least if the billable is inactive, we need telll user to create client first.

**Comment by jurre** [comment]
Maybe we could hide the button until you select a client?

**Comment by Marthyn** [comment]
I think the reason a project can't be billable without a client was a request from defacto. For now i agree with hiding the button until you select a client. 

Or what would be nice is that you can enter a new client already instead of having to select one in the dropdown.

**Comment by Danielshow** [comment]
Hi, I will like to contribute to this project, Have this been fixed?
I can work on it

**Comment by dkhgh** [comment]
> I think the reason a project can't be billable without a client was a request from defacto.

Still makes sense IMO.



> Hi, I will like to contribute to this project,

Cool.

> Have this been fixed?

It's not really clear if this needs work.

**Comment by Danielshow** [comment]
> It's not really clear if this needs work.
Ok that's fine then
Looking for where to start?
